internal/table/table.go

```diff
@@ -133,51 +140,120 @@ func rescaleColumns(colWidths []int, targetTotal int) {
+					pos++
+					break
+				}
+
+				buffer.WriteString(string(runeText[pos]))
+				pos++
+				count++
+			}
+		}
+		entries = append(entries, entry{buffer.String(), count})
+	}
+
+	// Clean up any entries that are just color tags
+	// This can happen when the end of a string or line
+	// break occurs right before a tag
+	for i, entry := range entries {
+		if entry.length != 0 {
+			continue
+		}
+
+		text := entries[i].line
+		if i >= 1 {
+			entries[i-1].line = entries[i-1].line + text
+		}
+		entries = entries[0:i]
+	}
+
+	return entries
+}
+
+func matchIndex(pos int, matches [][]int) (int, bool) {
+	for _, match := range matches {
+		start, stop := match[0], match[1]
+		if pos == start {
+			return stop, true
+		}
+	}
+	return -1, false
+}
+
 func renderRow(providedColumns []string, colWidths []int) string {
 	// don't want to modify the provided slice
 	columns := make([]string, len(providedColumns))
 	copy(columns, providedColumns)
 
 	result := ""
 
-	// Keep rendering lines until there's no column data left to render
-	for len(strings.Join(columns, "")) != 0 {
-		// Iterate over the columns by their line sizes
-		for n, maxLen := range colWidths {
-			// ignore columns that we do not have data for (they have been filled up with the last colValue already)
-			if len(columns) < n+1 {
-				continue
-			}
+	entries := make([][]entry, len(columns))
 
-			colValue := []rune(columns[n])
+	widths := make([]int, len(colWidths))
+	copy(widths, colWidths)
+	for n, maxLen := range colWidths {
+		// ignore columns that we do not have data for (they have been filled up with the last colValue already)
+		if len(columns) < n+1 {
+			continue
+		}
 
-			// Detect multi column span
-			if len(colWidths) > n+1 && len(columns) == n+1 {
-				for _, v := range colWidths[n+1:] {
-					maxLen += v
-				}
+		// Detect multi column span
+		if len(colWidths) > n+1 && len(columns) == n+1 {
+			for _, v := range colWidths[n+1:] {
+				maxLen += v
 			}
+		}
 
-			maxLen = maxLen - (padding * 2)
+		maxLen = maxLen - (padding * 2)
+		widths[n] = maxLen
 
-			// How much of the colValue are we using for this line?
-			end := len(colValue)
-			if end > maxLen {
-				end = maxLen
-			}
+		entries[n] = getCroppedText(columns[n], maxLen)
+	}
 
-			skipLineBreak := 0
-			if breakpos := runeSliceIndexOf(colValue, linebreakRune); breakpos != -1 && breakpos < end {
-				end = breakpos + 1
-				skipLineBreak = 1
-			}
+	totalRows := 0
+	for _, columnEntries := range entries {
+		if len(columnEntries) > totalRows {
+			totalRows = len(columnEntries)
+		}
+	}
 
-			suffix := strings.Repeat(" ", maxLen-end)
-			result += pad(string(colValue[0:end-skipLineBreak]) + suffix)
+	// Render each row
+	for i := 0; i < totalRows; i++ {
+		for n, columnEntry := range entries {
+			maxLen := widths[n]
+			text := ""
+			suffix := strings.Repeat(" ", maxLen)
+
+			if len(columnEntry) > i {
+				repeat := maxLen - columnEntry[i].length
+				suffix = strings.Repeat(" ", repeat)
```

Comment:
Should guard this against panics (if repeat is negative). Shouldn't happen, but I'd rather we have a slightly messed up looking table than a panic, in the event that there is a bug in this code.

Reply:
I originally had this, but removed it as I did not encounter it in any of the tests. I have added it back.